site/content/entry/yuzu-progress-report-may-2022/index.md

```diff
@@ -0,0 +1,272 @@
+
+Thankfully, we have a new system that can workaround those issues that are outside of our control.
+yuzu will now perform a Vulkan check at boot.
+
+If the check passes, yay!, you can use Vulkan or OpenGL and select which API to use, or in Vulkan’s case, which device to run yuzu with, as always from the Graphics section in configuration.
+
+{{< imgs
+    "./ok.png| Check passes, Vulkan works!"
+  >}}
+
+If this check fails, a warning will be displayed at the next time you launch yuzu. If this happens, you will only be able to use OpenGL as the graphics API. You will still have the option to pick the shader backend (GLSL, GLASM, SPIR-V) that best suits your needs.
+
+{{< imgs
+    "./error.png| Oh oh.."
+  >}}
+
+For those souls that happen to land in this situation, a button labeled "Check for Working Vulkan" at the bottom of the Graphics settings window will show up, allowing to retest Vulkan support.
+
+{{< imgs
+    "./button.png| Once you manage to solve the issue, click on the button at the bottom!"
+  >}}
+
+Thanks to [toastUnlimited](https://github.com/lat9nq), gone is OpenGL as the default graphics API. 
+Out with the old, in with the new. {{< gh-hovercard "8393" "Long live King Vulkan." >}}
+
+Going forward, our developers will focus their priority to Vulkan, but will continue to support OpenGL.
+OpenGL users are recommended to use the GLSL shader backend, as GLASM and SPIR-V will receive limited support from now on.
+
+## Graphical changes, driver issues, and the nostalgia bliss that is the good old 64
+
+This month, [byte[]](https://github.com/liamwhite) continues the wave of improvements for `Super Mario 3D All-Stars`, this time, he noticed a bug in the DMAcopy ([direct memory access](https://en.wikipedia.org/wiki/Direct_memory_access)) of the Nintendo Switch’s GPU.
+
+`DMACopy` is a mechanism that many games use to send texture data to the GPU, it handles the format conversion from "pitch" (pixels on a line by line basis) to "tiled" (gridded) images.
+This process works by writing the pitch image data into GPU memory accessible by the DMA engine. Next, a DMAcopy is requested through the DMA engine driver, converting the image data into a separate buffer accessible by the GPU. This buffer will then be used as the texture on the final draw.
+
+{{< gh-hovercard "8313" "After fixing `bytes_per_pixel`," >}} `Super Mario Galaxy` now has proper lens flare.
+
+{{< single-title-imgs-compare
+    "RTX On? (Super Mario Galaxy)"
+    "./dmabug.png"
+    "./dmafix.png"
+>}}
+
+byte[] also improved the way OpenGL interprets face flips depth, [replacing the previously reported fix](https://yuzu-emu.org/entry/yuzu-progress-report-apr-2022/#saving-princess-peach-yet-again). The previous implementation had terrible peformance in OpenGL, making it virtually unplayable.
```

Comment:
The current implementation has poor performance in _Vulkan_. OpenGL runs at full speed, which was the motivation to fix the rendering in OpenGL.

Reply:
Also the me getting face flips wrong the first time is just because it was tricky to test due to being an extremely uncommon configuration for the GPU.